manifests/service.pp

```diff
@@ -312,8 +325,10 @@
 
   case $service_provider {
     'systemd': {
-      file { '/etc/systemd/system/docker.service.d':
-        ensure => 'directory',
+      if $manage_systemd_override_dirs {
```

Comment:
I think those directories should not be created here at all  instead the puppet/systemd module should be used

Reply:
I agree, puppetlabs-docker should make use of the facilities provided by puppet-systemd in such cases. I was hesitant to propose or implement that because it is probably a larger change, and adds a new dependency.

Reply:
personally I would highly prefer that. maybe wait for feedback from @modules. I think the current implementation is really against best practices and adding the manage_systemd_override_dirs just makes it more challenging in the future to switch to puppet/systemd module without breaking changes

Reply:
Hi @antaflos and @bastelfreak. The Docker module leveraging puppet-systemd sounds like a sensible approach and we would happy to see a PR implementing such a change.